### ISSUE TITLE
Server stuck in collating errors

### ISSUE THREAD

**Issue by Liinkiing** [facebook/flow#7598]
Flow version: Starting from **0.93.0**

## Expected behavior
When I run **flow**, the flow server should correctly parse my files and then it should verify my codebase
## Actual behavior
Since v0.93.0 (0.92.1 works fine), when I run **flow**, flow is stuck after parsing my files and show this message for a while :
`Please wait. Server is handling a request (🗄  collating errors)`
And then my computer starts to get very high CPU / RAM usages. I've tried to let it do its things for a while, and it took 970~s to complete, whereas in version **0.92.1**, it took only **15s**.

I've tried with an empty **.flowconfig** to make sure that my config was not the problem, but it changed nothing.

(in case, here is what my **.flowconfig** looks like)

```
[version]
0.93.0

[lints]
all=warn

[ignore]
.*/node_modules/*
.*/bower_components/*
; ignore compiled files
.*/web/*
.*/vendor/*
.*/app/Resources/assets/*
.*/app/Resources/fonts/*
.*/bower.json
.*/.storybook/*
.*/docs/*
.*/coverage/*
.*/codemod/*

[libs]
flow-typed

[options]
emoji=true
module.use_strict=true


experimental.const_params=true

esproposal.export_star_as=enable

esproposal.decorators=ignore

;include_warnings=true
```

As for flow logs, I have nothing relevant, just stuff like 
```
[2019-03-28 15:23:17.131] Calculating dependencies
[2019-03-28 15:23:17.137] Merging
[2019-03-28 15:23:24.527] Merge skipped 0 of 1452 modules
[2019-03-28 15:23:24.545] Done
[2019-03-28 15:23:24.545] Checked set: Focused: 1452, Dependents: 0, Dependencies: 0
```
And then it is stuck at collating errors.

I don't know how to fix this, I've searched accross similar issues, but there don't seems to be many cases (I've found #133 but it's outdated).

**TL;DR**, my flow verification time increased from **15s** in **v0.92.1** to **970~s** in **v0.93.0**, while massively increasing my RAM usage and CPU

**Comment by avikchaudhuri** [comment]
Collating errors can run into some pathological cases for large errors. Not sure how to proceed, though...

**Comment by chapati23** [comment]
we're also noticing a pretty heavy performance drop since somewhere between 0.92 and 0.96

**Comment by fabiomcosta** [comment]
I'm on the same situation, `collating errors` takes about 30 secs after any file change I do in my codebase.
More info:
* Flow `0.97`
* I'm not using lazyMode
* Our project has about 800 modules using Flow and multiple other dependencies

**Comment by fabiomcosta** [comment]
@avikchaudhuri is there any way I could debug this myself and provide more information?
By looking at some a specific log maybe?

**Comment by fabiomcosta** [comment]
I can confirm that on our project the perf regression also starts happening on 0.93.0.
0.92.1 is noticeably faster.

**Comment by SamChou19815** [comment]
There are not enough information to repro, and it's likely that the issue has already been addressed.
